src/cargo/core/source/mod.rs

```diff
@@ -116,7 +116,10 @@ pub enum QueryKind {
     /// Each source gets to define what `close` means for it.
     /// Path/Git sources may return all dependencies that are at that URI,
     /// whereas an `Index` source may return dependencies that have the same canonicalization.
-    Fuzzy,
+    Alternatives,
+    /// Behavior like `Exact` for path sources and like `Alternatives` for
+    /// registry sources
```

Comment:
Please focus on the desired behavior, and not the implementation, for the comment.

This one will match a dependency in all ways except it will normalize the package name.  Each source defines what normalizing means.